packages/core/src/client/index.ts

```diff
@@ -1,31 +1,110 @@
 import type { Schema } from "@/internal/types.js";
 import type { ReadonlyDrizzle } from "@/types/db.js";
+import type { QueryWithTypings } from "drizzle-orm";
 import { createMiddleware } from "hono/factory";
+import { streamSSE } from "hono/streaming";
+
+const POLL_INTERVAL = 500;
 
 export const client = ({ db }: { db: ReadonlyDrizzle<Schema> }) => {
+  let id = 0;
+
   return createMiddleware(async (c, next) => {
-    if (c.req.path !== "/client") {
-      return next();
+    // @ts-ignore
+    const session: PgSession = db._.session;
+    const liveQueries = new Map<
+      number,
+      {
+        queryHash: string;
+        resultHash: string;
+      }
+    >();
+
+    // TODO(kyle) method?
+
+    if (c.req.path === "/client/db") {
+      const body = await c.req.json();
+
+      // @ts-ignore
+      const res = await session
+        .prepareQuery(
+          {
+            sql: body.sql,
+            params: body.params,
+            // @ts-ignore
+            typings: body.typings,
+          },
+          undefined,
+          undefined,
+          body.method === "all",
+        )
+        .execute();
+
+      // @ts-ignore
+      return c.json({ rows: res.rows.map((row) => Object.values(row)) });
     }
 
-    const body = await c.req.json();
+    if (c.req.path === "/client/live") {
+      c.header("Content-Type", "text/event-stream");
+      c.header("Cache-Control", "no-cache");
+      c.header("Connection", "keep-alive");
 
-    // @ts-ignore
-    const res = await db._.session
-      .prepareQuery(
-        {
-          sql: body.sql,
-          params: body.params,
-          // @ts-ignore
-          typings: body.typings,
+      const streamId = id++;
+
+      return streamSSE(
+        c,
+        async (stream) => {
+          stream.onAbort(() => {
+            stream.close();
+          });
+
+          while (stream.closed === false) {
+            const query = JSON.parse(c.req.query("query")!) as QueryWithTypings;
```

Comment:
Should probably return a 4xx status code if the `query` query param is not present or invalid. And should do this before you enter `streamSSE` I think.